src/webgui/src/webgui/server.py

```diff
@@ -26,10 +32,38 @@ def __init__(self, name):
 def index():
     return render_template('index.html', sub=sub)
 
-# Sample callback for ROS to use
-def callback(data):
-    print(data.data)
+@app.route('/l_camera')
+def l_camera():
+	return Response(video_feed(l_cam_topic), mimetype='multipart/x-mixed-replace; boundary=frame')
 
+@app.route('/r_camera')
+def r_camera():
+	return Response(video_feed(r_cam_topic), mimetype='multipart/x-mixed-replace; boundary=frame')
+
+@app.route('/b_camera')
+def b_camera():
+	return Response(video_feed(b_cam_topic), mimetype='multipart/x-mixed-replace; boundary=frame')
+
+def cvt_imgmsg_to_cv2(data, bridge):
+	try:
+		cv2_img = bridge.imgmsg_to_cv2(data, '8UC3')
+		ret, jpeg = cv2.imencode('.jpg', cv2_img)
+		if jpeg is not None:
+			return (b'--frame\r\n Content-Type: image/jpeg\r\n\r\n' + jpeg.tobytes() + b'\r\n')
+		else:
+			rospy.logerr('Failed to convert img msg to cv2:\n%s', 'Jpeg is none')
+
+
+	except CvBridgeError as re:
+		rospy.logerr('Failed to convert img msg to cv2:\n%s', e)
+	
+def video_feed(topic):
```

Comment:
Did you follow a specific article to do this? If so, can you put a link to it for others to go find?

Reply:
Also can we move this to a separate file that can act as a library for image handling?

Reply:
Which functions should be moved to a separate file?

Reply:
All the ones related to building the video feed. I'd like to just have routes in this file